src/render-delaying-services.js

```diff
@@ -56,11 +76,25 @@ const LOAD_TIMEOUT = 3000;
  *     as the detected render delaying services
  */
 export function waitForServices(win) {
-  const promises = includedServices(win).map(service => {
+  const promises = includedServices(win).map(serviceId => {
+
+    const serviceReadyPromise = getServicePromise(
+        win,
+        serviceId
+    ).then(service => {
+      if (service && service.whenReady) {
```

Comment:
Don't all services have to have a `whenReady`?

Reply:
Nope, Services are just JS classes. Though, we chose the name `whenReady` to match the naming scheme of other classes

Reply:
Sorry, don't all render blocking services have a `whenReady`?

Reply:
Ah yes they should. The reason why I am doing this though, is more for legacy reasons. I've gone and updated the old services to use the interface. But I may be a little redundant here. I can go ahead and remove this, and update all the tests so that they match this new assumption.

What do you think? 😄